projects/packages/jetpack-mu-wpcom/bin/sync-newspack-blocks.sh

```diff
@@ -128,30 +130,79 @@ mkdir -p $TARGET/shared
-cp -R $CODE/src/blocks/homepage-articles $TARGET/blocks/
-cp -R $CODE/src/blocks/carousel $TARGET/blocks/
-cp -R $CODE/src/shared $TARGET/
-cp -R $CODE/src/components $TARGET/
+cp "$CODE"/includes/class-newspack-blocks-api.php $TARGET/
+cp "$CODE"/includes/class-newspack-blocks.php $TARGET/
+cp -R "$CODE"/src/blocks/homepage-articles $TARGET/blocks/
+cp -R "$CODE"/src/blocks/carousel $TARGET/blocks/
+cp -R "$CODE"/src/shared $TARGET/
+cp -R "$CODE"/src/components $TARGET/
 
 # Get Typescript working by copying the main type defs over.
-cp $CODE/src/types/index.d.ts $TARGET/types/
+cp "$CODE"/src/types/index.d.ts $TARGET/types/
 # Function types need to be capitalized in our system. We only match " function"
 # beginning with a space to avoid matching it as a substring. (Not perfect, but
 # imperfections will be caught by CI with failing tsc, etc.)
 sed "${sedi[@]}" -e "s| function| Function|g" "$TARGET/types/index.d.ts"
 
 # Note: I would have used eslint-nibble, but it doesn't support autofixing via the CLI.
 echo "Changing JS textdomain to match jetpack-mu-wpcom..."
-pnpm --package=eslint@8.57.0 dlx eslint --no-ignore --rule '"@wordpress/i18n-text-domain":["error",{"allowedTextDomain":"jetpack-mu-wpcom"}]' --fix $TARGET > /dev/null
+BASE=$(cd "$(dirname "${BASH_SOURCE[0]}")"/../../../.. && pwd)
+FULLTARGET="$PWD/$TARGET"
+
+# Add a temporary single-rule eslint.config.mjs file.
+cat > "$TARGET/eslint.config.mjs" <<EOF
+import makeBaseConfig from 'jetpack-js-tools/eslintrc/base.mjs';
+
+// This directory is copy-pasted from elsewhere, but we still need to run this one rule over it so we can't just .eslintignore.
+export default [
+	// Import base config, but no rules.
+	...makeBaseConfig( import.meta.url ).map( block => ( { ...block, rules: {} } ) ),
+	// Enable just this one rule.
+	{
+		rules: {
+			"@wordpress/i18n-text-domain": [ "error", { allowedTextDomain: "jetpack-mu-wpcom" } ],
+		}
+	},
+];
+EOF
+( cd "$BASE" && pnpm run lint-file --no-inline-config --no-ignore --fix "$FULLTARGET" )
+rm "$TARGET/eslint.config.mjs"
 
 echo "Changing JS translation function call to avoid bad minification..."
 pnpm --package=jscodeshift dlx jscodeshift -t ./bin/sync-newspack-blocks-formatter.js --extensions=js $TARGET
 
+# Add temporary PHPCS config file.
+cat > "$TARGET/.phpcs.dir.xml" <<EOF
+<?xml version="1.0"?>
+<ruleset>
+	<rule ref="MediaWiki.Usage.ForbiddenFunctions">
+		<exclude name="MediaWiki.Usage.ForbiddenFunctions.isset"/>
+	</rule>
+</ruleset>
+EOF
 echo "Changing PHP textdomain to match jetpack-mu-wpcom..."
-../../../vendor/bin/phpcbf --standard=./.phpcs.dir.xml --filter=../../../vendor/automattic/jetpack-phpcs-filter/src/PhpcsFilter.php --runtime-set jetpack-filter-no-ignore -q $TARGET
+"$BASE"/vendor/bin/phpcbf --standard=./.phpcs.dir.xml --filter="$BASE"/vendor/automattic/jetpack-phpcs-filter/src/PhpcsFilter.php --runtime-set jetpack-filter-no-ignore -q "$TARGET"
+rm "$TARGET/.phpcs.dir.xml"
```

Comment:
May as well name the file like `phpcs.tmp.xml` instead.

Also I think the command here is a bit overcomplicated. If you just write the few rules you need all into `phpcs.tmp.xml`, you should be able to do just
```
"$BASE"/vendor/bin/phpcbf --standard=phpcs.tmp.xml "$TARGET"
```

Reply:
Started poking at this. Apparently `phpcs` doesn't like `phpcs.tmp.xml` as a filename. 😖

Reply:
I've simplified this in 8465bfc4da using the `WordPress.Utils.I18nTextDomainFixer` rule. However, it doesn't appear to catch/change instances in `wp_set_script_translations`. Thoughts?

Reply:
Looks like it needed `Jetpack.Functions.I18n` as well. Solved in 5d3d690.